spec.html

```diff
@@ -24176,16 +24314,265 @@ <h1>Runtime Semantics: Evaluation</h1>
+          1. Set _decorationState_.[[Finished]] to *true*.
+          1. If _kind_ is ~field~, then
+            1. If IsCallable(_initializer_) is *true*, append _initializer_ to _elementRecord_.[[Initializers]].
+            1. Else if _initializer_ is not *undefined*, throw a *TypeError* exception.
+          1. Else if _kind_ is ~accessor~, then
+            1. If _newValue_ is an Object, then
+              1. Let _newGetter_ be ? Get(_newValue_, *"get"*).
+              1. If IsCallable(_newGetter_) is *true*, set _elementRecord_.[[Get]] to _newGetter_.
+              1. Else if _newGetter_ is not *undefined*, throw a *TypeError* exception.
+              1. Let _newSetter_ be ? Get(_newValue_, *"set"*).
+              1. If IsCallable(_newSetter_) is *true*, set _elementRecord_.[[Set]] to _newSetter_.
+              1. Else if _newSetter_ is not *undefined*, throw a *TypeError* exception.
+              1. Let _initializer_ be ? Get(_newValue_, *"init"*).
+              1. If IsCallable(_initializer_) is *true*, append _initializer_ to _elementRecord_.[[Initializers]].
+              1. Else if _initializer_ is not *undefined*, throw a *TypeError* exception.
+            1. Else if _newValue_ is not *undefined*, throw a *TypeError* exception.
+          1. Else,
+            1. If IsCallable(_newValue_) is *true*, then
+              1. Perform MakeMethod(_newValue_, _homeObject_).
+              1. If _kind_ is ~getter~, then
+                1. Perform SetFunctionName(_newValue_, _key_, *"get"*).
+                1. Set _elementRecord_.[[Get]] to _newValue_.
+              1. Else if _kind_ is ~setter~, then
+                1. Perform SetFunctionName(_newValue_, _key_, *"set"*).
+                1. Set _elementRecord_.[[Set]] to _newValue_.
+              1. Else,
+                1. Perform SetFunctionName(_newValue_, _key_).
+                1. Set _elementRecord_.[[Value]] to _newValue_.
+            1. Else if _newValue_ is not *undefined*, throw a *TypeError* exception.
+        1. Set _elementRecord_.[[Decorators]] to ~empty~.
+        1. Return ~unused~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-applydecoratorstoclassdefinition" type="abstract operation">
+      <h1>
+        ApplyDecoratorsToClassDefinition (
+          _classDef_: a function object,
+          _decorators_: a List of function objects,
+          _extraInitializers_: a List of function objects,
+        ): either a normal completion containing a function object, or an abrupt completion
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>Applies the passed decorators to the passed class definition.</dd>
+      </dl>
+      <emu-alg>
+        1. For each element _decorator_ of _decorators_, do
+          1. Let _decorationState_ be the Record { [[Finished]]: *false* }.
+          1. Let _context_ be CreateDecoratorContextObject(~class~, _className_, _classExtraInitializers_, _decorationState_).
```

Comment:
The _className_ variable isn't defined anywhere in this AO.

Reply:
Neither is _classExtraInitializers_. Should that be a reference to the _extraInitializers_ parameter?